### DISCUSSION TITLE
NextJS meta tags issue

### DISCUSSION THREAD

**Discussion by rohan-pareek** [vercel/next.js#32894] [Help]
Hey everyone,

There is an issue I am facing regarding meta tags. I am actually trying to share a page (by pasting the page url inside social media post) that contains title, description and an image. I have added all the required meta tags inside `<Head>` tag. All these tags are visible in "Inspect Element". However, there are no meta tags visible when "View Source" is opened.

This result in not showing of social cards (for example twitter).

Here are the meta tags I have added - 

```
<Head>
<title>{title}</title>
<meta property="og:type" content="website" />
<meta property="og:title" content={title} />
<meta property="og:description" content={description} />
<meta property="og:url" content={url} />
<meta property="og:site_name" content={'My App'} />
<meta property="og:image" content={thumbnail} />
<meta name="twitter:card" content="summary" />
<meta name="twitter:card" content="summary_large_image" />
<meta name="twitter:description" content={description} />
<meta name="twitter:title" content={title} />
<meta name="twitter:image" content={thumbnail} />
</Head>
```

I have tried all the possible ways such as client side rendering, server side rendering. But still no luck.

PS: I am able to see all the meta tags in "View Source" if I add them inside `_document.js` file. But, as I need dynamic meta tags, I cannot add them inside `_document.js` file.

Please add you inputs.

Thanks in advance!

**Comment by akanz** [comment]
Could you solve the problem? I'm having the exact same issue at the moment

**Reply by raul-saez** [reply]
Maybe you could try to set those meta tags in `_app.js` passing them as props inside the pages that need dynamic tags. You should be able to use those props inside of your `App` then. I haven't tested this issue but just thinking of a possible workaround for you.
